### PR TITLE
Feature/Get Map Index

### PR DESCRIPTION
**Pull request by yYaniv**

### CODE REVIEW

File: UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/IndicesProvider.swift [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    
+    func handleArea(_ bounds: MGLCoordinateBounds) {
+        if let sTimer = timer {
+            sTimer.invalidate()
+            timer = nil
+        }
+        
+        timer = Timer.scheduledTimer(withTimeInterval: 0.3, repeats: false, block: { [unowned self] _ in
+            self.timer = nil
+            self.requestIndex(for: bounds)
```
Comment (yterletskyi) [inline UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/IndicesProvider.swift]: won't it accumulate old requests and then send all of them? We need to cancel all previous requests once new timer starts

File: UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/Utils/IndicesStorage.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    //MARK: - Public Interface
+
+extension IndicesStorage {
+    
+    func contains(_ element: MapIndex) -> Bool {
+        return indicies.contains(where: {$0.index == element.index })
+    }
+    
+    func append(_ index: MapIndex) {
+        if MapIndexValidator.isValid(index.zoom) {
```
Comment (yterletskyi) [inline UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/Utils/IndicesStorage.swift]: `MapIndexValidator.isValid` should be used not only for appending into a cache, but also as a filter for further operations (requesting pits)

File: UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/Utils/IndicesStorage.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            indicies.append(index)
```
Comment (yterletskyi) [inline UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/Utils/IndicesStorage.swift]: let's better `append` only once we have drawn pits

File: UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/IndicesProvider.swift (hunk 1 above)
Comment (yYaniv) [reply UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/IndicesProvider.swift]: Nope, bc it'll be invalidated in line 38 (attempt to handle area before prev one requested)

File: UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/Utils/IndicesStorage.swift (hunk 2 above)
Comment (yYaniv) [reply UARoads_swift/UARoads_swift/Classes/Controllers/Routes/Mapbox/Index/Utils/IndicesStorage.swift]: Agree, but let's leave as it is. i'd like to merge this PR and update storage behavior in next one, where real client of indices storage will be used. Ok?

**Comment by yYaniv** [comment]
@yterletskyi updated. Plz confirm

**Pull request merged**